core/trie/snap_support.go

```diff
@@ -0,0 +1,168 @@
+package trie
+
+import (
+	"github.com/NethermindEth/juno/core/felt"
+)
+
+func (t *Trie) IterateAndGenerateProof(startValue *felt.Felt, consumer func(key, value *felt.Felt) (bool, error),
```

Comment:
So effectively this function checks if there are leaves to the right of startQuery (which is the first key), and assuming there (ie, finished=false), it returns a left proof, and a right proof (if the lastKey is provided). If there are no more keys to the right of startQuery, then we don't return any proofs.

Reply:
And the logic to determine whether there are any leaves to the right of startQuery depends on the doIterate, and consumer functions? Imo, these are a bit confusing, and I'm wondering if there is a simpler way to determine if there are keys to the right. For example, couldn't we just check if any of the right hashes along the path from the root to the startQuery leaf are non-zero, and if so set finished=false (neglecting the edge case where we end up at the final leaf)